packages/diagrams/backend/sirius-components-diagrams/src/main/java/org/eclipse/sirius/components/diagrams/ViewModifier.java

```diff
@@ -0,0 +1,49 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.diagrams;
+
+import java.util.Collection;
+import java.util.Optional;
+
+/**
+ * All possible modifiers to define the state of a diagram element.
+ *
+ * @author tgiraudet
+ */
+public enum ViewModifier {
+
+    Hidden, Faded, Normal;
+
+    public static final ViewModifier DEFAULT_MODIFIER = ViewModifier.Normal;
+
+    /**
+     * Compute the dominant modifier the given collection where Hidden > Faded > Normal.
+     * @param modifiers
+     *            the modifier collection from which extract the dominant modifier
+     * @return the dominant modifier of the collection or {@link Optional#empty()} if the collection is empty
+     */
+    public static Optional<ViewModifier> getDominantModifier(Collection<ViewModifier> modifiers) {
+        if (modifiers.isEmpty())
+            return Optional.empty();
+
+        final ViewModifier modifier;
+        if (modifiers.contains(ViewModifier.Hidden))
+            modifier = ViewModifier.Hidden;
+        else if (modifiers.contains(ViewModifier.Faded))
+            modifier = ViewModifier.Faded;
+        else
+            modifier = ViewModifier.DEFAULT_MODIFIER;
+        return Optional.of(modifier);
+    }
```

Comment:
Do not create global methods like that. I am quite surprised that:

- This is not formatted properly
- This does not create an issue during the build

I'll update both the checkstyle rules and the PR to trigger the build to confirm this

Reply:
This kind of code is a good test for our future removal of the formatter save action, I'll add some missing checks in our Checkstyle configuration.